speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/frontend/IndicatorConfig.cpp

```diff
@@ -49,12 +52,63 @@ std::vector<tIndicatorData> IndicatorConfig::GetActiveIndicators(InterventionTyp
+/// @return                   The vector of indicator data
+std::vector<tIndicatorData> IndicatorConfig::GetNeutralIndicators(InterventionType p_interventionType)
+{
+    // Guard when no signals are to be sent, always return an empty vector.
+    if (p_interventionType == INTERVENTION_TYPE_NO_SIGNALS) return {};
+
+    // For the indicators that are active, get for the indicators that are neutral
+    for (const tIndicatorData& indicator : GetActiveIndicators(p_interventionType))
+    {
+        // This if-statement is build around the idea that either braking or steering is active, not both
+        if (indicator.Action == INTERVENTION_ACTION_BRAKE_NONE || indicator.Action == INTERVENTION_ACTION_BRAKE || indicator.Action == INTERVENTION_ACTION_ACCELERATE)
+        {
+            // Only steering is in neutral
+            tIndicatorData m_neutralSteer = IndicatorConfig::GetInstance()->GetNeutralIndicator(INTERVENTION_ACTION_STEER_NONE);
+            m_neutralIndicators = {m_neutralSteer};
+        }
+        else if (indicator.Action == INTERVENTION_ACTION_STEER_NONE || indicator.Action == INTERVENTION_ACTION_TURN_LEFT || indicator.Action == INTERVENTION_ACTION_TURN_RIGHT)
+        {
+            // Only braking is in neutral
+            tIndicatorData m_neutralBrake = IndicatorConfig::GetInstance()->GetNeutralIndicator(INTERVENTION_ACTION_BRAKE_NONE);
+            m_neutralIndicators = {m_neutralBrake};
+        }
+        else
+        {
+            // Both braking and steering are in neutral
+            tIndicatorData m_neutralSteer = IndicatorConfig::GetInstance()->GetNeutralIndicator(INTERVENTION_ACTION_STEER_NONE);
+            tIndicatorData m_neutralBrake = IndicatorConfig::GetInstance()->GetNeutralIndicator(INTERVENTION_ACTION_BRAKE_NONE);
+            m_neutralIndicators = {m_neutralSteer, m_neutralBrake};
+        }
+    }
+    return m_neutralIndicators;
+}
+
+/// @brief          Returns an element containing indicator data for only a neutral indicator
+/// @return         The element of neutral indicator data
+tIndicatorData IndicatorConfig::GetNeutralIndicator(InterventionAction action)
+{
+    std::vector<tIndicatorData> m_neutralIndicator;
+    for (const tIndicatorData& indicator : IndicatorConfig::GetInstance()->GetIndicatorData())
+    {
+        if (indicator.Action == action)
+            m_neutralIndicator.insert(m_neutralIndicator.end(), indicator);
+    }
+    tIndicatorData first = m_neutralIndicator.front();
+    return first;
+}
+
 /// @brief          Activates the given intervention indicator
 /// @param p_action The intervention to activate the indicators for
 void IndicatorConfig::ActivateIndicator(InterventionAction p_action)
 {
     // TODO: add to the vector instead of overwriting it, this also requires
     //       a way to remove the indicator after some time has passed.
+    std::vector<tIndicatorData> m_indicatorData = IndicatorConfig::GetInstance()->GetIndicatorData();
```

Comment:
Not sure what this is trying to achieve, GetIndicatorData returns m_indicatorData which you have access to directly from here. Also a local variable should not have the m_ prefix.